src/main/java/org/mifos/connector/channel/interceptor/IdInterceptor.java

```diff
@@ -0,0 +1,144 @@
+package org.mifos.connector.channel.interceptor;
+
+import com.fasterxml.jackson.annotation.JsonAutoDetect;
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.databind.DeserializationFeature;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.SerializationFeature;
+import com.fasterxml.jackson.datatype.jsr310.JavaTimeModule;
+import org.mifos.connector.channel.camel.routes.ChannelRouteBuilder;
+import org.mifos.connector.common.channel.dto.PhErrorDTO;
+import org.mifos.connector.common.exception.PaymentHubError;
+import org.mifos.connector.common.exception.PaymentHubErrorCategory;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.Primary;
+import org.springframework.data.redis.core.RedisTemplate;
+import org.springframework.stereotype.Component;
+import org.springframework.web.servlet.HandlerInterceptor;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.concurrent.TimeUnit;
+
+import static org.mifos.connector.channel.camel.config.CamelProperties.CLIENTCORRELATIONID;
+
+@Component
+@Primary
+public class IdInterceptor  implements HandlerInterceptor {
+
+    @Autowired
+    ChannelRouteBuilder channelRouteBuilder;
+
+    @Value("${redis.idempotency.enabled}")
+    Boolean redisIdempotencyEnabled;
+
+    @Value("${redis.idempotency.apiList}")
+    String apiList;
+
+    @Value("${redis.idempotency.keyFormat}")
+    String keyFormat;
+
+    @Value("${redis.cacheRetencyDuration")
+    Long cacheRetencyDuration;
+
+    @Autowired
+    public RedisTemplate<String, String> redisTemplate;
+
+    private final Logger logger = LoggerFactory.getLogger(this.getClass());
+
+    @Override
+    public boolean preHandle(HttpServletRequest request, HttpServletResponse response, Object handler)
+            throws Exception {
+
+        if (redisIdempotencyEnabled) {
+           Set<String> listParts = new HashSet<>(Arrays.asList(apiList.split(",")));
```

Comment:
This should be done at the constructor level, ie, this operation should happen only once during the startup. the  set should be a class-level (member) variable. so that it retains its value is in memory throughout. you can also make sure that this bean is a Singleton.

Reply:
sure